core-sdk/src/main/java/com/ably/tracking/LocationUpdateModels.kt

```diff
@@ -117,6 +119,60 @@ data class Location(
      */
     val time: Long
 ) {
+    /**
+     * Utility function that coerces non-finite values to sensible defaults
+     * where possible.
+     */
+    fun sanitize(): Location {
+        return Location(
+            latitude,
+            longitude,
+            altitude,
+            if (accuracy.isFinite()) accuracy else 0.0f,
+            if (bearing.isFinite()) bearing else 0.0f,
+            if (speed.isFinite()) speed else 0.0f,
+            time,
+        )
+    }
+
+    /**
+     * Utility function that returns a new, valid Location with no non-finite members.
+     * If the validation errors list is empty, the returned Location is non-null and valid; otherwise it is null.
+     */
+    fun validate(): Pair<Location?, ArrayList<LocationValidationError>> {
+        val errs: ArrayList<LocationValidationError> = ArrayList<LocationValidationError>()
```

Comment:
In Kotlin we tend to use lists instead of arrays, so I'd change this to a `MutableList` 😉 Also I think the less abbreviations the better so I'd change the name to `errors` or even `validationErrorMessages` 🤔 
```suggestion
        val errors: MutableList<LocationValidationError> = mutableListOf()
```